### DISCUSSION TITLE
A problem with VSG1.1.0

### DISCUSSION THREAD

**Discussion by Shanyzz** [vsg-dev/VulkanSceneGraph#1135] [Q&A] [answered]
Hello to the VSG Community,

I used the previously released version of vsg to build my project, read a cad model and manually create a plane to observe the shadows, everything worked fine(Figure 1).
![1](https://github.com/vsg-dev/VulkanSceneGraph/assets/120642500/d546cb2c-f58a-4cfb-a23f-21b13bc1f48b)

But today I updated the vsg version (1.1.0), as well as the vsgXchange and vsgQT versions. Things went wrong after the update, when I  executed the same code, the result turned into what is seen in Figure 2, where the plane shows a strange gradient effect that changes with the viewing angle（Figure 3）, and there were also some issues with the shadows. Initially, I thought it was the shaders I wrote that caused this issue, but even after replacing the shaders with standard.vert and standard_phong.frag from vsgexamples, the problem did not go away. 
![2](https://github.com/vsg-dev/VulkanSceneGraph/assets/120642500/2e8b2dd3-e6b8-4e61-8b61-1b650006d750)
![3](https://github.com/vsg-dev/VulkanSceneGraph/assets/120642500/6702ee57-4555-4de4-b5c0-deee7373c3bd)

I can't find the cause of all this, could you offer me some help?

The code for creating a plane is as follows:
`auto plane_colors = vsg::vec4Value::create(vsg::vec4{1.0, 1.0, 1.0, 1.0f});
auto plane_mat = vsg::PhongMaterialValue::create();
plane_mat->value().ambient.set(0.24725, 0.1995, 0.0745, 1.0f);
plane_mat->value().diffuse.set(0.75164, 0.60648, 0.22648, 1.0f);
plane_mat->value().specular.set(0.628281, 0.555802, 0.366065, 1.0f);
plane_mat->value().shininess = 100;

  float plane_length = 5000.0f;
  auto plane_vertices = vsg::vec3Array::create(
      {{-plane_length, -plane_length / 10, -plane_length},
       {plane_length, -plane_length / 10, -plane_length},
       {plane_length, -plane_length / 10, plane_length},
       {-plane_length, -plane_length / 10, plane_length}});

  auto plane_normals = vsg::vec3Array::create(
      {{0.0f, 1.0f, 0.0f},
       {0.0f, 1.0f, 0.0f},
       {0.0f, 1.0f, 0.0f},
       {0.0f, 1.0f, 0.0f}});

  auto plane_texcoords = vsg::vec2Array::create(
      {{0.0f, 0.0f},
       {1.0f, 0.0f},
       {1.0f, 1.0f},
       {0.0f, 1.0f}});

  auto plane_indices = vsg::ushortArray::create(
      {
          0,
          1,
          2,
          2,
          3,
          0,
      });

    vsg::DataList Env_vertexArrays;
    auto Env_graphicsPipelineConfig = vsg::GraphicsPipelineConfigurator::create(phong);     Env_graphicsPipelineConfig->assignArray(Env_vertexArrays, "vsg_Vertex", VK_VERTEX_INPUT_RATE_VERTEX, plane_vertices);
    Env_graphicsPipelineConfig->assignArray(Env_vertexArrays, "vsg_Normal", VK_VERTEX_INPUT_RATE_VERTEX, plane_normals);
    Env_graphicsPipelineConfig->assignArray(Env_vertexArrays, "vsg_TexCoord0", VK_VERTEX_INPUT_RATE_VERTEX, plane_texcoords);
    Env_graphicsPipelineConfig->assignArray(Env_vertexArrays, "vsg_Color", VK_VERTEX_INPUT_RATE_INSTANCE, plane_colors);
    Env_graphicsPipelineConfig->assignDescriptor("material", plane_mat);
    // 
    auto Env_drawCommands = vsg::Commands::create();
    Env_drawCommands->addChild(vsg::BindVertexBuffers::create(Env_graphicsPipelineConfig->baseAttributeBinding, Env_vertexArrays));
    Env_drawCommands->addChild(vsg::BindIndexBuffer::create(plane_indices));
    //cout << mesh.indexes->size() << endl;
    Env_drawCommands->addChild(vsg::DrawIndexed::create(6, 1, 0, 0, 0));

    //
    auto Env_rs = vsg::RasterizationState::create();
    Env_rs->cullMode = VK_CULL_MODE_NONE;
    Env_graphicsPipelineConfig->pipelineStates.push_back(Env_rs);

    Env_graphicsPipelineConfig->init();

    // create StateGroup as the root of the scene/command graph to hold the GraphicsPipeline, and binding of Descriptors to decorate the whole graph
    auto Env_stateGroup = vsg::StateGroup::create();
    Env_graphicsPipelineConfig->copyTo(Env_stateGroup);
    // set up model transformation node
    auto Env_transform = vsg::MatrixTransform::create(); //
    //transform->subgraphRequiresLocalFrustum = false;
    // add drawCommands to StateGroup
    Env_stateGroup->addChild(Env_drawCommands);
    Env_transform->addChild(Env_stateGroup);
    Env_scenegraph->addChild(Env_transform); `

**Comment by robertosfield** [comment, marked as answer]
I don't know what you are doing with each of the views, the code snippet you provide is just for a tiny part of what you are doing.

The best can do is provide general suggestions:

1. Try VSG & vsgXchange master - I'll tag 1.1.2 from this during the coming week.  master contains lots of extra features and fixes.
2. Run your application with Vulkan debug/validation layer enabled to see if this highlights any errors.
3. Try using vsg::Builder to create the geometry base layer.
4. Try output subrgraphs to .vsgt and load individually using examples like vsgviewer and vsgshadow to see what happens, use -d to enable the Vulkan debug/validation layer.
